app/src/main/java/com/practicum/playlistmaker/mediateka/ui/adapter/MediatekaViewPagerAdapter.kt

```diff
@@ -0,0 +1,25 @@
+package com.practicum.playlistmaker.mediateka.ui.adapter
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.Lifecycle
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import com.practicum.playlistmaker.mediateka.ui.fragment.FavoritesFragment
+import com.practicum.playlistmaker.mediateka.ui.fragment.PlaylistsFragment
+
+class MediatekaViewPagerAdapter(
+    fragmentManager: FragmentManager,
+    lifecycle: Lifecycle
+) : FragmentStateAdapter(fragmentManager, lifecycle) {
+
+    override fun getItemCount(): Int = 2
+
+    override fun createFragment(position: Int): Fragment {
+        return when (position) {
+            0 -> FavoritesFragment.newInstance()
+            1 -> PlaylistsFragment.newInstance()
+            else -> throw IllegalArgumentException("Invalid position: $position")
```

Comment:
В данном случае лучше от эксепшена лучше отказаться, так как в базовом классе этот эксепшн не обрабатывается и возможности такой нет, т.к. данный класс из Android SDK. Такой эксепшн приведёт к крешу в рантайме, чего лучше не допускать, обойти можно следующим образом:
```suggestion
            else -> PlaylistsFragment.newInstance()
```